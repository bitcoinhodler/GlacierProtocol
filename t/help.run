#!/bin/bash
set -e

$GLACIERSCRIPT --help
$GLACIERSCRIPT entropy --help
$GLACIERSCRIPT create-deposit-data --help
$GLACIERSCRIPT create-withdrawal-data --help
