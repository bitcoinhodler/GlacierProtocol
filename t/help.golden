usage: glacierscript.py [-h] [-v]
                        {entropy,create-deposit-data,create-withdrawal-data}
                        ...

optional arguments:
  -h, --help            show this help message and exit
  -v, --verbose         increase output verbosity

Subcommands:
  {entropy,create-deposit-data,create-withdrawal-data}
    entropy             Generate computer entropy
    create-deposit-data
                        Create cold storage address
    create-withdrawal-data
                        Construct withdrawal transaction

For more help, include a subcommand, e.g. `./glacierscript.py entropy --help`
usage: glacierscript.py entropy [-h] [--num-keys NUM_KEYS] [-r RNG]

optional arguments:
  -h, --help           show this help message and exit
  --num-keys NUM_KEYS  The number of keys to create random entropy for
  -r RNG, --rng RNG    Minimum number of 8-bit bytes to use for computer
                       entropy when generating private keys (default: 20)
usage: glacierscript.py create-deposit-data [-h] [-m M] [-n N] [-d DICE]
                                            [--p2wsh] [-r RNG]

optional arguments:
  -h, --help            show this help message and exit
  -m M                  Number of signing keys required in an m-of-n multisig
                        address creation (default m-of-n = 1-of-2)
  -n N                  Number of total keys required in an m-of-n multisig
                        address creation (default m-of-n = 1-of-2)
  -d DICE, --dice DICE  The minimum number of dice rolls to use for entropy
                        when generating private keys (default: 62)
  --p2wsh               Generate p2wsh (native segwit) deposit address,
                        instead of p2wsh-in-p2sh
  -r RNG, --rng RNG     Minimum number of 8-bit bytes to use for computer
                        entropy when generating private keys (default: 20)
usage: glacierscript.py create-withdrawal-data [-h]

optional arguments:
  -h, --help  show this help message and exit
